### DISCUSSION TITLE
Shorten Posts button suggestion

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#170] [Suggestion Box]
Standard disclaimer: I may not be in the primary audience for this script - I know some people are passionate about their shortened posts, and I'm pretty ambivalent! However:

Trying it out for a bit, I like it and could see myself leaving it on, but I repeatedly find my eyes jumping over the white expand button and not realizing the post is shortened, then getting confused. Perhaps coloring it just a bit, for example to the color of the box in ask posts, could increase this clarity without going so far as to look like a non-native element?

![image](https://user-images.githubusercontent.com/8336245/123919239-99b3e900-d939-11eb-86ed-4954caa6dad0.png)

Might still be a bit too subtle, I dunno! I played around with some box shadows, because if you know me, you know [I love to overuse box shadows](https://github.com/new-xkit/XKit/pull/1968), but nothing seemed great.

(As an aside, the bottom left corner of the shorten posts button isn't rounded on my Firefox. Why? No clue! It's fine in Chrome. Hopefully no one sees this and loses the ability to unsee it :P)

**Comment by AprilSylph** [comment]
Implemented in 838caf9b535c33cd5928db6a046ebf95f720516a! I went with the secondary accent colour, since that's (seemingly) designed to contrast subtley but obviously with the post background.

| Dark Mode | circa 2016 |
| ----------- | ---------- |
| ![image](https://user-images.githubusercontent.com/28949509/123969720-63fe1700-d9b0-11eb-87e0-ac7e51c36b22.png) | ![image](https://user-images.githubusercontent.com/28949509/123969822-7a0bd780-d9b0-11eb-8493-decf11c83367.png) |

Also, the bottom left border radius thing was probably caused by using `overflow-y: clip;`, a value which Chrome doesn't support (and so falls back to `overflow-y: hidden;`). This was so user avatars would stay visible, but behaviour across browsers being consistent is probably more important, so I've changed this to simply be `overflow: hidden;` in 1f2639ae444b71744841db906a85cbe8d9c79b2f. (Using just `overflow-y: hidden;` caused horizontal scrollbars to appear on Firefox 😱)

**Reply by marcustyphoon** [reply]
Love the button! Looks native to me ❤️ 

Oof, the user avatar thing is unfortunate. My brain aliases posts that lack an avatar to "ad container," as well as the impact to e.g. Seen Posts' "only dim avatars" option.

If it's just the three pixels, I'd be inclined to say `overflow-y: clip` is worth it. Also, was this just... a thing, when using Rewritten's Shorten Posts in Chrome? (I mean, I guess I can test that... and yes. Yikes. But the XKit 7 way of modifying the avatar was pretty hard to reverse in a `clean()`... hm.)

**Reply by marcustyphoon** [reply]
Heyyy, wait a second... just using `overflow-y: clip` is supported in Chrome ≥90; it doesn't fallback... Buuuuut it appears to behave exactly as `overflow-y: hidden`. Great job, Chrome!

(Also tried it in Chromium latest and it still doesn't work.)

**Reply by marcustyphoon** [reply]
~~omg it was [that](/AprilSylph/XKit-Rewritten/commit/f6ffb57776c129d611327328af70e8211bb8a611) easy?!?!~~ edit: oh I see. well. firefox best browser, anywayyy

I wonder how hard it is to submit a bug to chromium(?)

**Reply by AprilSylph** [reply]
that didn't fix it on chromium? cringe

**Reply by marcustyphoon** [reply]
oh it did! ain't pretty but it works. just gotta remove the `top: 69px;` on the `xkit-shorten-posts-shortened _3qGyx` ("avatar") on chromium, presumably

<img width="300" alt="Screen Shot 2021-07-02 at 10 53 18 AM" src="https://user-images.githubusercontent.com/8336245/124312246-d7746580-db23-11eb-9d56-fb2cb3fbe9f1.png">

**Reply by AprilSylph** [reply]
1. guess i'll do that on monday
2. why are your screenshots always so tiny

**Reply by marcustyphoon** [reply]
https://jsfiddle.net/w60sr9k7/1/ hmm... this works fine in chrome... there must be some subtlety I missed that causes the sticky container to vanish.

2. I always upload them in full retina resolution, have them take up like half the page height, then think, "this seems like a waste, the thing I want to communicate is visible at the proverbial 10,000 feet," then make them smaller :P

**Reply by marcustyphoon** [reply]
[ignore the last 5 messages](https://github.com/AprilSylph/XKit-Rewritten/pull/176)

**Reply by AprilSylph** [reply]
> I always upload them in full retina resolution [...] then make them smaller

i'd strongly prefer you simply take up all that page height, or only resize down to ~1080p. i like seeing things.

**Reply by marcustyphoon** [reply]
Sure thing, can do!
